### DISCUSSION TITLE
Using an IDE (here visual studio code) with brainvisa singularity

### DISCUSSION THREAD

**Discussion by JoelChavas** [brainvisa/brainvisa.github.io#71] [Show and tell]
Sometimes, to use a debugger or to follow libraries (soma.aims, and others), you can use an IDE from the host system that has the right access to all brainvisa stuff.

For example, to use VS code including all brainvisa singularity power, you need to perform the following steps (we suppose you have a brainvisa singularity image installed, as per https://brainvisa.info/web/download.html): 

- Install the IDE VS code on the ubuntu host system
- Find the location of the VS code binary on the host (the true location, not the symbolic link pointing to it): here, we suppose that the VS code binary is located at /usr/share/code/bin/code
- The IDE will launch properly with brainvisa properly integrated when using the following command:
`bv opengl=nv /host/usr/share/code/bin/code`

**Comment by denisri** [comment]
This will work only if the host system is compatible with the image system (ubuntu 18.04 for now, 20.4 soon), but in this situation, yes it is probably a good solution. Thanks for the hint.

Otherwise there have been demands to provide VScode in the dev image (brainvisa/casa-distro#269), we haven't decided yet if this is a good idea. The thing is that different people will want to use different IDEs, there are several ones (for instance VScode, PyCharm, Eclipse, QtCreator, and so on) and we cannot install all of them in the image.

**Comment by rbregand** [comment]
When VScode is launched from bv environment, the debugger is working fine (stop correctly at breakpoints). We wanted VScode in the dev image but the tests just show it's not recquired. 

Performed on Mint 20.1
